frontend/src/app/main/data/changes.cljs

```diff
@@ -178,19 +178,23 @@
   (ptk/reify ::commit-changes
     ptk/WatchEvent
     (watch [_ state _]
-      (let [file-id  (or file-id (:current-file-id state))
-            uchg     (vec undo-changes)
-            rchg     (vec redo-changes)
-            features (features/get-team-enabled-features state)]
-
-        (rx/of (-> params
-                   (assoc :undo-group undo-group)
-                   (assoc :features features)
-                   (assoc :tags tags)
-                   (assoc :stack-undo? stack-undo?)
-                   (assoc :save-undo? save-undo?)
-                   (assoc :file-id file-id)
-                   (assoc :file-revn (resolve-file-revn state file-id))
-                   (assoc :undo-changes uchg)
-                   (assoc :redo-changes rchg)
-                   (commit)))))))
+      (let [file-id      (or file-id (:current-file-id state))
+            uchg         (vec undo-changes)
+            rchg         (vec redo-changes)
+            features     (features/get-team-enabled-features state)
+            user-viewer? (not (get-in state [:workspace-file :permissions :can-edit]))]
```

Comment:
Prefer, dm/get-in over get-in when the vector of traversal path is known statically
Also:
- it is ok for a first approach, but I think we need to start to abstract the permission checking to named funcions

Reply:
I'm not sure... The permissions are in different places for dashboard and workspace. I've tried  a few things, but it becomes a little convoluted. I propose don't make this messy, and address the abstraction when we need it